library/ssl_tls13_generic.c

```diff
@@ -2740,29 +2740,31 @@ static int ssl_finished_in_postprocess( mbedtls_ssl_context* ssl )
 }
 
 #if defined(MBEDTLS_ZERO_RTT)
-void mbedtls_ssl_set_early_data( mbedtls_ssl_context *ssl, int early_data,
-                                 char *buffer, unsigned int len,
-                                 int(*early_data_callback)( mbedtls_ssl_context *,
-                                                            unsigned char *, size_t ) )
+void mbedtls_ssl_conf_early_data( mbedtls_ssl_config* conf, int early_data, unsigned int max_early_data,
+                                  int(*early_data_callback)( mbedtls_ssl_context*,
+                                                             unsigned char*, size_t ))
 {
 #if !defined(MBEDTLS_SSL_SRV_C)
     ( (void ) early_data_callback );
 #endif /* !MBEDTLS_SSL_SRV_C */
-
-    if( ssl != NULL )
+    if( conf != NULL )
     {
-        ssl->early_data_enabled = early_data;
-        if( buffer != NULL && len >0 && early_data==MBEDTLS_SSL_EARLY_DATA_ENABLED )
-        {
-            ssl->early_data_buf = buffer;
-            ssl->early_data_len = len;
+        conf->early_data_enabled = early_data;
+
 #if defined(MBEDTLS_SSL_SRV_C)
+        conf->max_early_data = max_early_data;
```

Comment:
This seems a bit inconsistent: Why always set `max_early_data` but set the callback only if `max_early_data > 0 ` and 0-RTT is enabled? 

I'd suggest to set _both_ of `max_early_data` and `early_data_callback` precisely if `early_data == MBEDTLS_SSL_EARLY_DATA_ENABLED`.